src/group.ts

```diff
@@ -339,50 +340,58 @@ export class Group {
+            let tv1 = v.square() //         1. tv1 = v^2
+            const tv2 = u.mul(v) //         2. tv2 = u * v
+            tv1 = tv1.mul(tv2) //           3. tv1 = tv1 * tv2
+            let y1 = tv1.power(c1) //       4. y1 = tv1^c1
+            y1 = y1.mul(tv2) //             5. y1 = y1 * tv2
+            const y2 = y1.mul(c2) //        6. y2 = y1 * c2
+            let tv3 = y1.square() //        7. tv3 = y1^2
+            tv3 = tv3.mul(v) //             8. tv3 = tv3 * v
+            const isQR = tv3.equals(u) //   9. isQR = tv3 == u
+            const y = cmov(y2, y1, isQR) // 10. y = CMOV(y2, y1, isQR)
+            return { isQR, root: y } //     11. return (isQR, y)
+        }
 
-        let tv1 = u.square() //          1. tv1 = u^2
-        const tv3 = Z.mul(tv1) //        2. tv3 = Z * tv1
-        let tv2 = tv3.square() //       3. tv2 = tv3^2
-        let xd = tv2.add(tv3) //        4.  xd = tv2 + tv3
-        let x1n = xd.add(one) //         5. x1n = xd + 1
-        x1n = x1n.mul(B) //              6. x1n = x1n * B
-        let tv4 = p.sub(A)
-        xd = xd.mul(tv4) //              7.  xd = -A * xd
-        const e1 = xd.equals(zero) //    8.  e1 = xd == 0
-        tv4 = A.mul(Z)
-        xd = cmov(xd, tv4, e1) //        9.  xd = CMOV(xd, Z * A, e1)
-        tv2 = xd.square() //            10. tv2 = xd^2
-        const gxd = tv2.mul(xd) //      11. gxd = tv2 * xd
-        tv2 = tv2.mul(A) //             12. tv2 = A * tv2
-        let gx1 = x1n.square() //       13. gx1 = x1n^2
-        gx1 = gx1.add(tv2) //           14. gx1 = gx1 + tv2
-        gx1 = gx1.mul(x1n) //           15. gx1 = gx1 * x1n
-        tv2 = gxd.mul(B) //             16. tv2 = B * gxd
-        gx1 = gx1.add(tv2) //           17. gx1 = gx1 + tv2
-        tv4 = gxd.square() //           18. tv4 = gxd^2
-        tv2 = gx1.mul(gxd) //           19. tv2 = gx1 * gxd
-        tv4 = tv4.mul(tv2) //           20. tv4 = tv4 * tv2
-        let y1 = tv4.power(c1) //       21.  y1 = tv4^c1
-        y1 = y1.mul(tv2) //             22.  y1 = y1 * tv2
-        const x2n = tv3.mul(x1n) //     23. x2n = tv3 * x1n
-        let y2 = y1.mul(c2) //          24.  y2 = y1 * c2
-        y2 = y2.mul(tv1) //             25.  y2 = y2 * tv1
-        y2 = y2.mul(u) //               26.  y2 = y2 * u
-        tv2 = y1.square() //            27. tv2 = y1^2
-        tv2 = tv2.mul(gxd) //           28. tv2 = tv2 * gxd
-        const e2 = tv2.equals(gx1) //  29.  e2 = tv2 == gx1
-        const xn = cmov(x2n, x1n, e2) //  30.  xn = CMOV(x2n, x1n, e2)
-        let y = cmov(y2, y1, e2) //     31.   y = CMOV(y2, y1, e2)
-        const e3 = sgn(u) === sgn(y) // 32.  e3 = sgn0(u) == sgn0(y)
-        tv1 = p.sub(y)
-        y = cmov(tv1, y, e3) //         33.   y = CMOV(-y, y, e3)
-        let x = xd.inverseMod(p) //     34. return (xn, xd, y, 1)
-        x = xn.mul(x)
-
-        const point = new sjcl.ecc.point(this.curve, new sjcl.bn(x), new sjcl.bn(y))
```

Comment:
I'm having trouble finding where this maps to the spec. Could you add a link?

Reply:
added a link at the top of function